docs/discovery.md

```diff
@@ -42,10 +42,34 @@ class MyClass
 }
 ```
 
+## HTTP Async Client Discovery
+
+This type of discovery finds a HTTP Async Client implementation.
+
+``` php
+use Http\Client\HttpAsyncClient;
+use Http\Discovery\HttpAsyncClientDiscovery;
+
+class MyClass
+{
+    /**
+     * @var HttpAsyncClient
+     */
+    protected $httpAsyncClient;
+
+    /**
+     * @param HttpAsyncClient|null $httpAsyncClient to do HTTP requests.
```

Comment:
Hm...you are right, we should explain why this parameter is optional. Is it explained where the HttpClient is documented?